### PR TITLE
[feature] [null support # 5] Support null in scalar transform function wrapper

### PR DESCRIPTION
**Pull request by 61yao**
Support null in scalar transform.
Scalar function invocation will determine whether the result is null or not.

See the github issue:
https://github.com/apache/pinot/issues/10252

Design doc:
https://docs.google.com/document/d/1w2eBuZrS73xE2JIw8MYlySy5pNaX9iE8kZr_YAL20CM/edit

### CODE REVIEW

**Comment by 61yao** [comment]
> nullableParameters

nullableParameters is still needed because when it is set to false, we return null if any of the argument is null:

You can check this file for details: https://github.com/apache/pinot/pull/10376/files#diff-688b9166dc63a0dbc1f1726deb4847284cc4f2ebbf9ccbab64c4631e68c33c23

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ScalarTransformFunctionWrapper.java [hunk 1]
```diff
@@ -85,14 +89,45 @@ public void init(List<TransformFunction> arguments, Map<String, ColumnContext> c
...
+    _scalarArguments = new Object[numArguments];
     _nonLiteralIndices = new int[numArguments];
     _nonLiteralFunctions = new TransformFunction[numArguments];
     for (int i = 0; i < numArguments; i++) {
       TransformFunction transformFunction = arguments.get(i);
       if (transformFunction instanceof LiteralTransformFunction) {
-        String literal = ((LiteralTransformFunction) transformFunction).getStringLiteral();
-        _arguments[i] = parameterTypes[i].convert(literal, PinotDataType.STRING);
+        LiteralTransformFunction literalTransformFunction = (LiteralTransformFunction) transformFunction;
+        DataType dataType = literalTransformFunction.getResultMetadata().getDataType();
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ScalarTransformFunctionWrapper.java:99]: Not introduced in this PR, but can literal be `INT` or `FLOAT` type? `NumberUtils.createNumber(literal)` will parse to the string to the smallest type, and we should probably keep this convention

Comment (KKcorps) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ScalarTransformFunctionWrapper.java:99]: Yes +1

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/LiteralTransformFunction.java [hunk 2]
```diff
@@ -68,24 +68,28 @@ public LiteralTransformFunction(LiteralContext literalContext) {
...
   }
 
   public int getIntLiteral() {
     return _intLiteral;
   }
 
+  public long getLongLiteral() {
+    return _longLiteral;
+  }
+
```
[the remark this answers is not in the document]
Comment (61yao) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/LiteralTransformFunction.java:82]: ditto.

**Review by KKcorps — APPROVED** [review]

**Comment by KKcorps** [comment]
https://github.com/apache/pinot/pull/10562#discussion_r1164780161

This will be taken care in another PR

**Pull request merged**